packages/ember-toucan-core/src/components/form/controls/autocomplete.ts

```diff
@@ -465,48 +415,41 @@ export default class ToucanFormAutocompleteControlComponent<
    *    option.
    */
   @action
-  resetValue(event: Event) {
+  resetEverything(event: Event) {
     assert(
       'Expected HTMLInputElement',
       event.target instanceof HTMLInputElement
     );
 
-    let { selected, optionKey, onChange } = this.args;
-
-    if (!selected) {
+    if (!this.args.selected) {
       return;
     }
 
     // Reset our filtered options as we are going to "clear" the input
-    this.filteredOptions = undefined;
+    this.filteredOptions = null;
 
     if (event.target.value === '') {
-      this.inputValue = undefined;
-      onChange?.(null);
-
-      return;
-    }
-
-    if (typeof selected === 'string' && event.target.value !== selected) {
-      this.inputValue = selected;
+      this.inputValue = '';
+      this.args.onChange?.(null);
 
       return;
     }
 
     if (
-      typeof selected === 'object' &&
-      optionKey &&
-      event.target.value !== selected[optionKey]
+      typeof this.args.selected === 'string' &&
```

Comment:
Is `typeof this.args.selected === 'string'` still needed?

Reply:
Nope! Nixed.